crates/matrix-sdk-crypto/src/backups/mod.rs

```diff
@@ -349,6 +350,44 @@ impl BackupMachine {
         }
     }
 
+    /// Sign a [`RoomKeyBackupInfo`] using the device's identity key and, if
+    /// available, the cross-signing master key.
+    ///
+    /// # Arguments
+    ///
+    /// * `backup_info`: The backup version that should be verified. Should
+    /// be created from the [`BackupDecryptionKey`] using the
+    /// [`BackupDecryptionKey::to_backup_info()`] method.
+    pub async fn sign_backup(
+        &self,
+        backup_info: &mut RoomKeyBackupInfo,
+    ) -> Result<(), SignatureError> {
+        if let RoomKeyBackupInfo::MegolmBackupV1Curve25519AesSha2(data) = backup_info {
```

Comment:
So we just silently do nothing for other variants?

Reply:
If we don't throw an error, I think there should at least be a log event emitted.

Reply:
Added an error here: https://github.com/matrix-org/matrix-rust-sdk/pull/2819/commits/ea6e105e2384f77520fa2441b5b6a2a0715a1071.